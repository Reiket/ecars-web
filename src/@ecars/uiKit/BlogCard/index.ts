import {BlogCardContent} from '@ecars/uiKit/BlogCard/components/BlogCardContent';
import {BlogCardComponent} from '@ecars/uiKit/BlogCard/BlogCardComponent';
import {BlogCardImage} from '@ecars/uiKit/BlogCard/components/BlogCardImage';

export const BlogCard = Object.assign(BlogCardComponent, {
  Content: BlogCardContent,
  Image: BlogCardImage,
});
