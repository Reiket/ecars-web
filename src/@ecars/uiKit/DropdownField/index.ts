import {DropdownFieldComponent} from '@ecars/uiKit/DropdownField/DropdownFieldComponent';
import {DropdownFieldWrapper} from '@ecars/uiKit/DropdownField/components/DropdownFieldWrapper';

export const DropdownField = Object.assign(DropdownFieldComponent, {
  Wrapper: DropdownFieldWrapper,
});
