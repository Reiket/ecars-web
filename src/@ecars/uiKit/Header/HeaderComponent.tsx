import type {FC} from 'react';
import {Header, Navbar} from '@ecars/uiKit/Header/index';
import {Button, Logo} from 'ecars-web-lib';
import {useBurger} from '@ecars/services/hooks/useBurger';

export const HeaderComponent: FC = () => {
  const {isBurgerActive, toggleBurger} = useBurger();
  return (
    <Header.Wrapper>
      <div className="header__body">
        <Header.Burger
          isBurger={isBurgerActive}
          onClick={toggleBurger}
        />
        <Logo src="./assets/images/logo-site/logo.svg" />
        <Navbar isBurgerActive={isBurgerActive} />
        <Header.Utils />
        <Button
          size="medium"
          color="green"
        >
          Get a quote
        </Button>
      </div>
    </Header.Wrapper>
  );
};
