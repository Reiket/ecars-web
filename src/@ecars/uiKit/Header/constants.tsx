import {Icons, withBlockClass} from 'ecars-web-lib';
import {HeaderWrapper} from '@ecars/uiKit/Header/components/HeaderWrapper';
import type {PageUrlType} from '@ecars/constants/page-urls';
import {PageUrls} from '@ecars/constants/page-urls';
import {HeaderNavbarWrapper} from '@ecars/uiKit/Header/components/HeaderNavbarWrapper';
import type {ReactNode} from 'react';

export const HeaderWrapperHOC = withBlockClass(HeaderWrapper, 'header');
export const HeaderNavbarWrapperHOC = withBlockClass(HeaderNavbarWrapper, 'navbar');

interface NavbarLink {
  to: PageUrlType;
  label: string;
  isUtil?: boolean;
}

interface HeaderUtil {
  to: PageUrlType;
  icon: ReactNode;
}

export const navbarLinksConfig: NavbarLink[] = [
  {
    to: PageUrls.CATALOG,
    label: 'All Cars',
  },
  {
    to: PageUrls.ABOUT,
    label: 'About Us',
  },
  {
    to: PageUrls.Blog,
    label: 'Blog',
  },
  {
    to: PageUrls.FAQ,
    label: 'FAQ',
  },
  {
    to: PageUrls.FAVORITES,
    label: 'Favorites',
    isUtil: true,
  },
  {
    to: PageUrls.LOGIN,
    label: 'Login',
    isUtil: true,
  },
];

export const headerUtilsConfig: HeaderUtil[] = [
  {to: PageUrls.FAVORITES, icon: <Icons.FavoriteSolid />},
  {to: PageUrls.LOGIN, icon: <Icons.Account />},
];

export const headerTestScenarios = [
  {
    desc: 'initial state',
    scrolls: [0],
    expected: {sticky: false, showHeader: true},
  },
  {
    desc: 'scrolled down below topbarHeight → sticky true, showHeader false',
    scrolls: [100],
    expected: {sticky: true, showHeader: false},
  },
  {
    desc: 'scroll up again → sticky true, showHeader true',
    scrolls: [150, 50],
    expected: {sticky: true, showHeader: true},
  },
];
