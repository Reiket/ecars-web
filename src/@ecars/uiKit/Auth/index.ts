import {AuthLayout} from '@ecars/uiKit/Auth/components/AuthLayout';
import {AuthPageWrapperHOC} from '@ecars/uiKit/Auth/constants';

export const AuthPage = Object.assign(AuthLayout, {
  Wrapper: AuthPageWrapperHOC,
});