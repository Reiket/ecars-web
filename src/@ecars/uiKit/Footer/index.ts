import {FooterComponent} from '@ecars/uiKit/Footer/FooterComponent';
import {FooterCopyWrapperHOC, FooterWrapperHOC} from '@ecars/uiKit/Footer/constants';
import {FooterCopy} from '@ecars/uiKit/Footer/components/FooterCopy';
import {FooterCopyList} from '@ecars/uiKit/Footer/components/FooterCopyList';
import {FooterColumn} from '@ecars/uiKit/Footer/components/FooterColumn';
import {FooterLogoBlock} from '@ecars/uiKit/Footer/components/FooterLogoBlock';
import {FooterLinkList} from '@ecars/uiKit/Footer/components/FooterLinkList';

export const Footer = Object.assign(FooterComponent, {
  Wrapper: FooterWrapperHOC,
  Column: FooterColumn,
  LogoBlock: FooterLogoBlock,
  LinkList: FooterLinkList,
});

export const Copy = Object.assign(FooterCopy, {
  Wrapper: FooterCopyWrapperHOC,
  List: FooterCopyList,
});
