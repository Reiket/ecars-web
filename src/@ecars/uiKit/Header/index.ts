import {HeaderComponent} from '@ecars/uiKit/Header/HeaderComponent';
import {HeaderNavbarWrapperHOC, HeaderWrapperHOC} from '@ecars/uiKit/Header/constants';
import {HeaderNavbar} from '@ecars/uiKit/Header/components/HeaderNavbar';
import {HeaderUtils} from '@ecars/uiKit/Header/components/HeaderUtils';
import {HeaderBurger} from '@ecars/uiKit/Header/components/HeaderBurger';
import {HeaderNavbarList} from '@ecars/uiKit/Header/components/HeaderNavbarList';

export const Header = Object.assign(HeaderComponent, {
  Wrapper: HeaderWrapperHOC,
  Utils: HeaderUtils,
  Burger: HeaderBurger,
});

export const Navbar = Object.assign(HeaderNavbar, {
  Wrapper: HeaderNavbarWrapperHOC,
  List: HeaderNavbarList,
});
