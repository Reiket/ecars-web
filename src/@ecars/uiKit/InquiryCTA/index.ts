import {InquiryCtaComponent} from '@ecars/uiKit/InquiryCTA/InquiryCtaComponent';
import {InquiryCtaWrapperHOC} from '@ecars/uiKit/InquiryCTA/constants';

export const InquiryCTA = Object.assign(InquiryCtaComponent, {
  Wrapper: InquiryCtaWrapperHOC,
});
