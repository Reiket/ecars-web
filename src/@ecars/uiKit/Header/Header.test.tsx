import {Header, Navbar} from '@ecars/uiKit/Header';
import {act, fireEvent, render, renderHook} from '@testing-library/react';

import {MemoryRouter} from 'react-router-dom';
import {useBurger} from '@ecars/services/hooks/useBurger';
import {useHeaderScroll} from '@ecars/services/hooks/useHeaderScroll';

const scenarios = [
  {
    desc: 'initial state',
    scrolls: [0],
    expected: {sticky: false, showHeader: true},
  },
  {
    desc: 'scrolled down below topbarHeight → sticky true, showHeader false',
    scrolls: [100],
    expected: {sticky: true, showHeader: false},
  },
  {
    desc: 'scroll up again → sticky true, showHeader true',
    scrolls: [150, 50],
    expected: {sticky: true, showHeader: true},
  },
];

vi.mock('@ecars/services/helpers/helpers', () => ({
  bodyOverflow: vi.fn(),
}));

vi.mock('@ecars/services/hooks/useBurger', () => ({
  useBurger: () => ({
    isBurgerActive: false,
    toggleBurger: toggleBurgerMock,
  }),
}));

const toggleBurgerMock = vi.fn();

describe('Header Component', () => {
  test('render component correctly', () => {
    const {container} = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('clicking burger toggles active class', () => {
    const toggleBurgerMock = vi.fn();
    const {container} = render(
      <MemoryRouter>
        <Header.Burger
          onClick={toggleBurgerMock}
          isBurger
        />
        <Navbar isBurgerActive />
      </MemoryRouter>,
    );
    const burger = container.querySelector('.burger');
    const navbar = container.querySelector('.navbar');
    if (burger) {
      fireEvent.click(burger);
    }
    expect(burger).toHaveClass('_active');
    expect(navbar).toHaveClass('_open-burger');
    expect(toggleBurgerMock).toHaveBeenCalledTimes(1);
  });
  test('renders and toggles burger hook', () => {
    const {container} = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    const burger = container.querySelector('.burger');
    if (burger) {
      fireEvent.click(burger);
    }
    expect(toggleBurgerMock).toHaveBeenCalledTimes(1);
  });
  scenarios.forEach((scenario) => {
    const {desc, scrolls, expected} = scenario;
    test(desc, () => {
      const {container} = render(
        <MemoryRouter>
          <Header />
        </MemoryRouter>,
      );
      const header = container.querySelector('header');

      act(() => {
        scrolls.forEach((y) => {
          window.scrollTo(0, y);
          window.dispatchEvent(new Event('scroll'));
        });
      });

      if (expected.sticky) {
        expect(header).toHaveClass('sticky');
      } else {
        expect(header).not.toHaveClass('sticky');
      }

      if (expected.showHeader) {
        expect(header).toHaveClass('show-header');
      } else {
        expect(header).not.toHaveClass('show-header');
      }
    });
  });
});

describe('useBurger hook', () => {
  test('returns the initial state as false by default', () => {
    const {result} = renderHook(() => useBurger());

    expect(result.current.isBurgerActive).toBe(false);
  });

  test('accepts a custom initial state', () => {
    const {result} = renderHook(() => useBurger(true));

    expect(result.current.isBurgerActive).toBe(true);
  });

  test('toggleBurger changes state and calls bodyOverflow', async () => {
    const {result} = renderHook(() => useBurger());
    const {bodyOverflow} = await import('@ecars/services/helpers/helpers');

    expect(result.current.isBurgerActive).toBe(false);

    act(() => {
      result.current.toggleBurger();
    });

    expect(result.current.isBurgerActive).toBe(true);
    expect(bodyOverflow).toHaveBeenCalledWith(true);

    act(() => {
      result.current.toggleBurger();
    });

    expect(result.current.isBurgerActive).toBe(false);
    expect(bodyOverflow).toHaveBeenCalledWith(false);
  });
});

describe('useHeaderScroll', () => {
  beforeEach(() => {
    window.scrollTo(0, 0);
  });

  test('should return initial state', () => {
    const {result} = renderHook(() => useHeaderScroll());
    expect(result.current.isSticky).toBe(false);
    expect(result.current.showHeader).toBe(true);
  });

  test('should set isSticky to true when scrolled below topbarHeight', () => {
    const {result} = renderHook(() => useHeaderScroll());

    act(() => {
      window.scrollTo(0, 100);
      window.dispatchEvent(new Event('scroll'));
    });

    expect(result.current.isSticky).toBe(true);
  });

  test('should hide header when scrolling down', () => {
    const {result} = renderHook(() => useHeaderScroll());

    act(() => {
      window.scrollTo(0, 10);
      window.dispatchEvent(new Event('scroll'));
    });

    expect(result.current.showHeader).toBe(false);
  });

  test('should show header when scrolling up', () => {
    const {result} = renderHook(() => useHeaderScroll());

    act(() => {
      window.scrollTo(0, 100);
      window.dispatchEvent(new Event('scroll'));
    });
    expect(result.current.showHeader).toBe(false);

    act(() => {
      window.scrollTo(0, 50);
      window.dispatchEvent(new Event('scroll'));
    });
    expect(result.current.showHeader).toBe(true);
  });
});
