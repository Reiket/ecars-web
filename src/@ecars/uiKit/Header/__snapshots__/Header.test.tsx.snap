// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > render component correctly 1`] = `
<div>
  <header
    class="header show-header"
  >
    <div
      class="header__container container container--lg"
    >
      <div
        class="header__body"
      >
        <button
          class="header__burger burger"
        >
          <span />
        </button>
        <a
          class="header__logo logo"
        >
          <img
            alt="Logo"
            src="./assets/images/logo-site/logo.svg"
          />
        </a>
        <nav
          class="header__navbar navbar"
        >
          <div
            class="navbar__overlay"
          />
          <ul
            class="navbar__list"
          >
            <li
              class="navbar__item"
            >
              <a
                class="nav-link"
                data-discover="true"
                href="/catalog"
              >
                All Cars
              </a>
            </li>
            <li
              class="navbar__item"
            >
              <a
                class="nav-link"
                data-discover="true"
                href="/about"
              >
                About Us
              </a>
            </li>
            <li
              class="navbar__item"
            >
              <a
                class="nav-link"
                data-discover="true"
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li
              class="navbar__item"
            >
              <a
                class="nav-link"
                data-discover="true"
                href="/faq"
              >
                FAQ
              </a>
            </li>
            <li
              class="navbar__item _hidden-util"
            >
              <a
                class="nav-link"
                data-discover="true"
                href="/favorites"
              >
                Favorites
              </a>
            </li>
            <li
              class="navbar__item _hidden-util"
            >
              <a
                class="nav-link"
                data-discover="true"
                href="/login"
              >
                Login
              </a>
            </li>
            <li
              class="navbar__item"
            >
              <div
                class="navbar__tools"
              >
                <div
                  class="navbar__dropdown-field dropdown-field"
                >
                  <p
                    class="dropdown-field__label"
                  >
                    Ship to:
                  </p>
                  <div
                    class="navbar__dropdown-menu dropdown-menu dropdown-menu--light"
                    data-testid="dropdownTestId"
                  >
                    <div
                      class="dropdown-menu__block"
                    >
                      <button
                        class="dropdown-menu__category"
                      >
                        <img
                          alt="ca"
                          src="./assets/images/flags/canada.svg"
                        />
                        Canada
                        <svg
                          class="dropdown-menu__arrow"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                            fill="none"
                          />
                          <path
                            d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                          />
                        </svg>
                      </button>
                    </div>
                    <ul
                      class="dropdown-menu__list"
                    >
                      <li
                        class="dropdown-menu__option _selected-option"
                        data-value="ca"
                      >
                        <img
                          alt="ca"
                          src="./assets/images/flags/canada.svg"
                        />
                         
                        Canada
                      </li>
                      <li
                        class="dropdown-menu__option"
                        data-value="us"
                      >
                        <img
                          alt="usa"
                          src="./assets/images/flags/usa.svg"
                        />
                         
                        USA
                      </li>
                      <li
                        class="dropdown-menu__option"
                        data-value="eu"
                      >
                        <img
                          alt="eu"
                          src="./assets/images/flags/europe.svg"
                        />
                         
                        Europe
                      </li>
                    </ul>
                  </div>
                </div>
                <div
                  class="navbar__dropdown-field dropdown-field"
                >
                  <p
                    class="dropdown-field__label"
                  >
                    Currency:
                  </p>
                  <div
                    class="navbar__dropdown-menu dropdown-menu dropdown-menu--light"
                    data-testid="dropdownTestId"
                  >
                    <div
                      class="dropdown-menu__block"
                    >
                      <button
                        class="dropdown-menu__category"
                      >
                        US Dollar
                        <svg
                          class="dropdown-menu__arrow"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                            fill="none"
                          />
                          <path
                            d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                          />
                        </svg>
                      </button>
                    </div>
                    <ul
                      class="dropdown-menu__list"
                    >
                      <li
                        class="dropdown-menu__option _selected-option"
                        data-value="usd"
                      >
                         
                        US Dollar
                      </li>
                      <li
                        class="dropdown-menu__option"
                        data-value="eur"
                      >
                         
                        Euro
                      </li>
                      <li
                        class="dropdown-menu__option"
                        data-value="gbp"
                      >
                         
                        British Pound
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </li>
            <li
              class="navbar__item"
            >
              <ul
                class="navbar__social social"
              >
                <li
                  class="social__item"
                >
                  <a
                    class="navbar__link link link--gray link--icon"
                    data-testid="routerLinkTestId"
                    href="https://www.facebook.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 320 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="social__item"
                >
                  <a
                    class="navbar__link link link--gray link--icon"
                    data-testid="routerLinkTestId"
                    href="https://twitter.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="social__item"
                >
                  <a
                    class="navbar__link link link--gray link--icon"
                    data-testid="routerLinkTestId"
                    href="https://www.youtube.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="social__item"
                >
                  <a
                    class="navbar__link link link--gray link--icon"
                    data-testid="routerLinkTestId"
                    href="https://www.instagram.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M512 378.7c-73.4 0-133.3 59.9-133.3 133.3S438.6 645.3 512 645.3 645.3 585.4 645.3 512 585.4 378.7 512 378.7zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zM512 717.1c-113.5 0-205.1-91.6-205.1-205.1S398.5 306.9 512 306.9 717.1 398.5 717.1 512 625.5 717.1 512 717.1zm213.5-370.7c-26.5 0-47.9-21.4-47.9-47.9s21.4-47.9 47.9-47.9 47.9 21.4 47.9 47.9a47.84 47.84 0 0 1-47.9 47.9z"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="social__item"
                >
                  <a
                    class="navbar__link link link--gray link--icon"
                    data-testid="routerLinkTestId"
                    href="https://wa.me/971558119024"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      data-testid="leftIconLinkTestId"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M260.062 32C138.605 32 40.134 129.701 40.134 250.232c0 41.23 11.532 79.79 31.559 112.687L32 480l121.764-38.682c31.508 17.285 67.745 27.146 106.298 27.146C381.535 468.464 480 370.749 480 250.232 480 129.701 381.535 32 260.062 32zm109.362 301.11c-5.174 12.827-28.574 24.533-38.899 25.072-10.314.547-10.608 7.994-66.84-16.434-56.225-24.434-90.052-83.844-92.719-87.67-2.669-3.812-21.78-31.047-20.749-58.455 1.038-27.413 16.047-40.346 21.404-45.725 5.351-5.387 11.486-6.352 15.232-6.413 4.428-.072 7.296-.132 10.573-.011 3.274.124 8.192-.685 12.45 10.639 4.256 11.323 14.443 39.153 15.746 41.989 1.302 2.839 2.108 6.126.102 9.771-2.012 3.653-3.042 5.935-5.961 9.083-2.935 3.148-6.174 7.042-8.792 9.449-2.92 2.665-5.97 5.572-2.9 11.269 3.068 5.693 13.653 24.356 29.779 39.736 20.725 19.771 38.598 26.329 44.098 29.317 5.515 3.004 8.806 2.67 12.226-.929 3.404-3.599 14.639-15.746 18.596-21.169 3.955-5.438 7.661-4.373 12.742-2.329 5.078 2.052 32.157 16.556 37.673 19.551 5.51 2.989 9.193 4.529 10.51 6.9 1.317 2.38.901 13.531-4.271 26.359z"
                      />
                    </svg>
                    +971 55 811 9024
                  </a>
                </li>
                <li
                  class="social__item"
                >
                  <a
                    class="navbar__link link link--gray link--icon"
                    data-testid="routerLinkTestId"
                    href="mailto:request@example.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      data-testid="leftIconLinkTestId"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                      />
                    </svg>
                    request@example.com
                  </a>
                </li>
              </ul>
            </li>
          </ul>
        </nav>
        <ul
          class="header__utils"
        >
          <li
            class="header__util"
          >
            <a
              class="link link--dark link--icon"
              data-discover="true"
              data-testid="routerLinkTestId"
              href="/favorites"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M923 283.6a260.04 260.04 0 0 0-56.9-82.8 264.4 264.4 0 0 0-84-55.5A265.34 265.34 0 0 0 679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 0 0-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                />
              </svg>
            </a>
          </li>
          <li
            class="header__util"
          >
            <a
              class="link link--dark link--icon"
              data-discover="true"
              data-testid="routerLinkTestId"
              href="/login"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2ZM12.1597 16C10.1243 16 8.29182 16.8687 7.01276 18.2556C8.38039 19.3474 10.114 20 12 20C13.9695 20 15.7727 19.2883 17.1666 18.1081C15.8956 16.8074 14.1219 16 12.1597 16ZM12 4C7.58172 4 4 7.58172 4 12C4 13.8106 4.6015 15.4807 5.61557 16.8214C7.25639 15.0841 9.58144 14 12.1597 14C14.6441 14 16.8933 15.0066 18.5218 16.6342C19.4526 15.3267 20 13.7273 20 12C20 7.58172 16.4183 4 12 4ZM12 5C14.2091 5 16 6.79086 16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5ZM12 7C10.8954 7 10 7.89543 10 9C10 10.1046 10.8954 11 12 11C13.1046 11 14 10.1046 14 9C14 7.89543 13.1046 7 12 7Z"
                />
              </svg>
            </a>
          </li>
        </ul>
        <button
          class="header__button button button--medium button--green"
          data-testid="buttonTestId"
          type="button"
        >
          Get a quote
        </button>
      </div>
    </div>
  </header>
</div>
`;
